docs/book/src/architecture/controllers/control-plane.md

```diff
@@ -0,0 +1,200 @@
+# Control Plane Controller
+
+![](../../images/control-plane-controller.svg)
+
+The Control Plane controller's main responsibilities are:
+
+* Managing a set of machines that represent a stacked control plane.
+* Provide information about the state of the control plane to downstream
+  consumers.
+
+A reference implementation is managed within the core Cluster API project as the
+Kubeadm control plane controller (`KubeadmControlPlane`). In this document,
+we refer to an example `ImplementationControlPlane` where not otherwise specified.
+
+## Contracts
+
+### Control Plane Provider
+
+The general expectation of a control plane controller is to instantiate a
+Kubernetes control plane consisting of the following services:
```

Comment:
This should likely list the services with links to their sections below.

Reply:
Probably should have said earlier. I've updated the PR